provider/diag/error.go

```diff
@@ -201,3 +201,39 @@ func FromError(err error, dt DiagnosticType, opts ...BaseErrorOption) Diagnostic
 		return Diagnostics{NewBaseError(err, dt, opts...)}
 	}
 }
+
+// Mutate generates new Diagnostics (using BaseError) from the given Diagnostics, and applies the given options to each one
+func Mutate(dd Diagnostics, opts ...BaseErrorOption) Diagnostics {
```

Comment:
where is this used? its kinda cool but still

Reply:
@roneli To distinguish between Configure diags vs. Fetch diags. See https://github.com/cloudquery/cloudquery/pull/594 and the last commit in it

Reply:
I don't think you need to set a resource name / id for diagnostics as its only for "resource diags", but I see the usage

Reply:
So we don't need this function basically, we can remove if not.